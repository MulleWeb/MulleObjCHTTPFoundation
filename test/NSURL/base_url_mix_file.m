//
//  main.m
//  NSURLTest
//
//  Created by Nat! on 24.02.20.
//  Copyright Â© 2020 Nat!. All rights reserved.
//
#ifdef __MULLE_OBJC__
# import <MulleObjCInetFoundation/MulleObjCInetFoundation.h>
# include <mulle-stacktrace/mulle-stacktrace.h>
#else
# import <Foundation/Foundation.h>
#endif

static void   run_test( NSURL *baseURL, NSString *string)
{
   NSURL      *URL;
   NSString   *result;

   URL = [NSURL URLWithString:string
                relativeToURL:baseURL];
   result = [URL absoluteString];
   if( ! URL && ! [NSURL URLWithString:string])
      result = @"relative is invalid";

   printf( "%s (base)\n + %s (relative)\n-> %s\n\n", [[baseURL absoluteString] UTF8String],
                                   [string UTF8String],
                                   [result UTF8String]);
}


static void   test( NSURL *baseURL)
{
   NSString   *string;

   string = @"file://"
             "host:32765"
             "/path";
   run_test( baseURL, string);

   string  = @"/rel_path/foo.html";
   run_test( baseURL, string);

   string = @"rel_path/bar.html";
   run_test( baseURL, string);
}


int   main(int argc, const char * argv[])
{
   NSURL   *baseURL;

#ifdef __MULLE_OBJC__
   struct _mulle_objc_universe   *universe;

   universe = mulle_objc_global_get_universe( __MULLE_OBJC_UNIVERSEID__);
   universe->debug.count_stackdepth = mulle_stacktrace_count_frames;
#endif

   baseURL = [NSURL URLWithString:@"file://"
                                  "base-host"
                                  "/base-path/"];
   test( baseURL);

   baseURL = [NSURL URLWithString:@"file://"
                                  "base-host"
                                  "/base-path"];
   test( baseURL);

   return 0;
}


