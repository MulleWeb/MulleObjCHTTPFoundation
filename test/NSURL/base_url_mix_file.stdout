file://base-host/base-path/ (base)
 + file://host:32765/path (relative)
-> file://host:32765/path

file://base-host/base-path/ (base)
 + /rel_path/foo.html (relative)
-> file://base-host/rel_path/foo.html

file://base-host/base-path/ (base)
 + rel_path/bar.html (relative)
-> file://base-host/base-path/rel_path/bar.html

file://base-host/base-path (base)
 + file://host:32765/path (relative)
-> file://host:32765/path

file://base-host/base-path (base)
 + /rel_path/foo.html (relative)
-> file://base-host/rel_path/foo.html

file://base-host/base-path (base)
 + rel_path/bar.html (relative)
-> file://base-host/rel_path/bar.html

